aea/helpers/search/models.py

```diff
@@ -230,13 +359,48 @@ def _check_validity(self):
                 if len(self.value) > 0:
                     _type = type(next(iter(self.value)))
                     assert all(isinstance(obj, _type) for obj in self.value)
+            elif self.type == ConstraintTypes.DISTANCE:
+                assert isinstance(self.value, (list, tuple))
+                assert len(self.value) == 2
+                assert isinstance(self.value[0], Location)
+                assert isinstance(self.value[1], float)
             else:
                 raise ValueError("Type not recognized.")
         except (AssertionError, ValueError):
             return False
 
         return True
 
+    def is_valid(self, attribute: Attribute) -> bool:
+        """
+        Check if the constraint type is valid wrt a given attribute.
+
+        :param attribute: the data model used to check the validity of the constraint type.
+        :return: ``True`` if the constraint type is valid wrt the attribute, ``False`` otherwise.
+        """
+        if self.type == ConstraintTypes.EQUAL:
+            return True
```

Comment:
@marcofavorito I didn't fill them in yet.. we should though

Reply:
Add an issue?

Reply:
Yes

Reply:
#1113